staging/src/k8s.io/apiserver/pkg/endpoints/filters/impersonation.go

```diff
@@ -116,6 +116,22 @@ func WithImpersonation(handler http.Handler, requestContextMapper request.Reques
 			}
 		}
 
+		if username != user.Anonymous {
+			// When impersonating a non-anonymous user,
+			// if neither the system:authenticated nor system:unauthenticated groups are explicitly included,
+			// include the system:authenticated group in the impersonated user info
+			found := false
+			for _, group := range groups {
+				if group == user.AllAuthenticated || group == user.AllUnauthenticated {
+					found = true
+					break
+				}
+			}
+			if !found {
+				groups = append(groups, user.AllAuthenticated)
```

Comment:
> I'm hard pressed to think of any scenario where that behavior would be helpful and not confusing.

If you don't do that, you're stealing power from the API.  I can see why you'd add the group for cases where its not found (same as the downstream "groupify for me"), but to take what the user requested and decide, "I'll add this too" (which we don't do downstream), seems wrong to me.